op-node/node/config.go

```diff
@@ -124,6 +126,11 @@ func (cfg *Config) Check() error {
 	if err := cfg.L2.Check(); err != nil {
 		return fmt.Errorf("l2 endpoint config error: %w", err)
 	}
+	if cfg.Beacon != nil {
```

Comment:
At some point, this cannot be `nil`. Should we enforce that its not `nil` if ecotone is active to prevent a panic on misconfiguration?

Reply:
made it error out if ecotone upgrade is active based on current time.

Reply:
also added a bit more error checking around nil beacon client